src/Builder.js

```diff
@@ -14,10 +14,43 @@ var pexprs = require('./pexprs');
 function Builder() {}
 
 Builder.prototype = {
+  currentDecl: null,
+
   newGrammar: function(name) {
     return new GrammarDecl(name);
   },
 
+  grammar: function(metaInfo, name, superGrammar, defaultStartRule, rules) {
+    var gDecl = new GrammarDecl(name);
+    if (superGrammar) {
+      gDecl.withSuperGrammar(this.fromRecipe(superGrammar));
+    }
+    if (defaultStartRule) {
+      gDecl.withDefaultStartRule(defaultStartRule);
+    }
+    if (metaInfo && metaInfo.source) {
+      gDecl.withSource(metaInfo.source);
+    }
+
+    var self = this;
+    this.currentDecl = gDecl;
+    Object.keys(rules).forEach(function(ruleName) {
+      var ruleRecipe = rules[ruleName];
+
+      var action = ruleRecipe[0]; // define/extend/override
+      // var metaInfo = ruleRecipe[1]; // TODO: us and handle meta-info
```

Comment:
Rules currently don't have meta-info attached. We could leave this out (and simplify the call below) or just go with the general format that `[1]` is the meta-info in JSON recipes and therefore keep it empty.
@pdubroy @alexwarth